code/modules/spells/ability_types/realized.dm

```diff
@@ -948,3 +949,190 @@
 		playsound(get_turf(H), "shatter", 50, TRUE)
 		to_chat(H, "<span class='userdanger'>Your pebble violently shatters!</span>")
 	return
+/* Flesh Idol - Repentance */
+/obj/effect/proc_holder/ability/prayer
+	name = "Prayer"
+	desc = "An ability that does causes you to start praying reducing damage taken by 50% but removing your ability to move and lowers justice by 80. \
+	When you finish praying everyone in a big area gets a 25% damage boost and gets healed."
+	action_icon_state = "flesh0"
+	base_icon_state = "flesh"
+	cooldown_time = 180 SECONDS
+
+/obj/effect/proc_holder/ability/prayer/Perform(target, mob/living/carbon/human/user)
+	ADD_TRAIT(user, TRAIT_IMMOBILIZED, type)
```

Comment:
You could add all TRAIT_IMMOBILIZED handling to the status effect and have the if statement for the `do_after` call `user.remove_status_effect(/datum/status_effect/flesh1)`